vmdb/app/models/ems_refresh.rb

```diff
@@ -181,16 +181,13 @@ def self.reconfig_save_vm_inventory(vm, hashes)
     child_keys = [:operating_system, :hardware]
     remove_keys = child_keys
 
-    # Backup keys that cannot be written directly to the database
-    key_backup = backup_keys(hash, remove_keys)
-
     begin
       raise MiqException::MiqIncompleteData if hash[:invalid]
 
       $log.info("#{log_header} Updating Vm [#{vm.name}] id: [#{vm.id}] location: [#{vm.location}] storage id: [#{vm.storage_id}] uid_ems: [#{vm.uid_ems}]")
-      vm.update_attributes!(hash)
+      vm.update_attributes!(hash.except(*remove_keys))
 
-      save_child_inventory(vm, key_backup, child_keys)
+      save_child_inventory(vm, hash, child_keys)
```

Comment:
Won't this send more information than is necessary?  Does it even matter, since child_keys is there?